### ISSUE TITLE
time travel does not work

### ISSUE THREAD

**Issue by richlv** [mapillary/mapillary_issues#2362]
firefox 49.0.2.
time travel was recently reintroduced, but it does not seem to work.
there's one image, one can select image to compare to - but nothing happens.
![mapillary-time-travel-not-working](https://cloud.githubusercontent.com/assets/112888/20237895/9c646f30-a8e7-11e6-8ee3-14568ed1d73f.png)

**Comment by oscarlorentzon** [comment]
Time travel works on most occations but when the original GPS positions are too far from each other the slider is not currently shown.

**Comment by richlv** [comment]
hmm, but these two seem to be extremely close :
https://www.mapillary.com/app/time-travel/?focus=photo&pKey=4gFK36TdzUZVCyoEO5F2lw&cKey=mhJtvKV0LXS7795ELnxdsQ

or is it not the original position that's shown on the map ?

try all the other images in that link - nothing happens for any of them...

**Comment by oscarlorentzon** [comment]
@richlv The original GPS positions for the compared photos in that link are not sufficiently close according to logic in `MapillaryJS 2.0`. That is descibed in the issue I linked to above.

See the original GPS positions on the map:

https://www.mapillary.com/app/?lat=56.97496099999998&lng=24.12830900000003&z=17&pKey=4gFK36TdzUZVCyoEO5F2lw&focus=map

https://www.mapillary.com/app/user/quicksand?lat=56.97496099999998&lng=24.12830900000003&z=17&pKey=mhJtvKV0LXS7795ELnxdsQ

**Comment by oscarlorentzon** [comment]
As a reference, it generally works like this:

https://www.mapillary.com/app/time-travel?lat=55.60583668000001&lng=13.006879879999985&z=17&pKey=rdg8JjR1q3as_P_uCDxmcA&cKey=jRJkrEoMzhDCWV4R72dxPQ&focus=photo

**Comment by oscarlorentzon** [comment]
We have made changes in MapillaryJS so that we fall back to a motionless transition instead of showing no slider at all when the distance between the photos based on the original GPS is too long.

Therefore it is now possible to compare the photos in the link you posted above.

**Comment by richlv** [comment]
hmm. right now nothing happens when the slider is used, but when it crosses 1/2 of the bar, the other image is suddenly displayed.
is that intended ?

**Comment by oscarlorentzon** [comment]
Yes, that is intended. It is a fallback when there is no smooth transition so the slider output is binary with a threshold at 0.5. The idea is that this behavior better for now than not showing a slider at all when the data is not fully trusted for smooth transitions.

Maybe you have noticed when navigating with the navigation arrows that there is not a smooth transition everywhere. The behavior is now the same for the slider.

**Comment by voschix** [comment]
For me Time Travel is still not working on Windows 10 and Firefox  (up-to-date and javascript enabled).

**Comment by oscarlorentzon** [comment]
@voschix Let's try to figure out why.

1. Can you share the examples where it does not work? 
2. Do those particular examples work in any other browsers or operating systems? 
3. What behavior are you expecting?
4. What behavior are you experiencing?

**Comment by voschix** [comment]
I have to rectract my statement. It works in many places. I was unlucky, I had tried it in about 10 random locations with multiple tracks of mine . Looking closer into it, I seem to have a number of sequences that are not normalized. These seem to be the problem. Those seem not to take parte in the Time <travel comparisons. Can you confirm that?
This opens another question: Why do I produce so many not-normalized sequences, when I always have my app looking "forward"? Is there an automatic method of detecting not-normalized sequences?

**Comment by nickplesha** [comment]
@voschix If you're talking about the time-travel options being available: these images are chosen based on the similarity of position (lat/long and compass angle) to the image you are on.

If all sequences on a location are "normalized" then there could be more overlapping images and these would show up in time-travel mode.

We don't have a way of detecting if a sequence is not normalized, unfortunately. 

The compass on the device you use to capture images might need calibrating to give more accurate measurements (e.g. https://www.youtube.com/watch?v=sP3d00Hr14o).

**Comment by voschix** [comment]
@nickplesha As I said, its the missing normalization in many of my sequences that interferes with time travel (when I first tested I did it in places where I had taken photos many times).
It's difficult to keep all one's sequences checked (normalization, removing bad photos, pixellation, position correction), when your are taking many photos. Not only is there no way to check for normalization, there is also no way of bookkeeping interventions in general for sequences.
I check the compass from time to time and do the occasional recalibration. I will do it more often in the future.
